lib/timezone/lookup/geonames.rb

```diff
@@ -36,11 +34,21 @@ def lookup(lat, long)
 
       private
 
+      def get_timezone_id(data)
+        return data['timezoneId'] if data['timezoneId']
+
+        if config.offset_etc_zones && data['gmtOffset']
+          return 'Etc/UTC' if data['gmtOffset'] == 0
+          "Etc/GMT#{format('%+d', data['gmtOffset'].to_i)}"
```

Comment:
Oh, it's just a safeguard in case API returns something other than a number. But maybe ArgumentError exception would be better if API returns unexpected data.